src/utils.js

```diff
@@ -1,22 +1,44 @@
 const CALLBACK_NAME = '_grecaptchaonloadcallback'
 
+let loadScriptPromise
+
+// Returns a promise that resolves once Google's reCAPTCHA library is loaded. If the library is already loaded, then no
+// work is performed, otherwise the library is dynamically loaded.
+// @param {string} [locale] - one of the language codes at https://developers.google.com/recaptcha/docs/language.
+// @returns {Promise}
 export function loadScript(locale) {
-  return new Promise((resolve, reject) => {
-    const url = `https://www.google.com/recaptcha/api.js?onload=${encodeURIComponent(
-      CALLBACK_NAME
-    )}${locale === false ? '' : `&hl=${encodeURIComponent(locale)}`}`
+  // Do this work just once, regardless of how many times this function is called, by saving the created promise and
+  // returning it forevermore.
+  if (!loadScriptPromise) {
+    // It is possible that the reCAPTCHA library has already been loaded by alternative means (perhaps with a static
+    // <script> tag in the HTML). In that case there's no work to do, so just return a pre-resolved promise. Otherwise
+    // dynamically load the script.
+    if (window.grecaptcha) {
```

Comment:
Here we could add an return to remove the else statement. 

```js
if (window.grecaptcha) {
  loadScriptPromise = Promise.resolve()
  return loadScriptPromise
}
```

Reply:
What if we got rid of the if statements altogether? This is how it would look (and how the linter wants it formatted):

```javascript
export function loadScript(locale) {
  // It is possible that the reCAPTCHA library has already been loaded by alternative means (perhaps with a static
  // <script> tag in the HTML). In that case there's no work to do, so just return a pre-resolved promise. Otherwise
  // dynamically load the script. Do this work just once, regardless of how many times this function is called, by
  // saving the created promise and returning it forevermore.
  loadScriptPromise =
    loadScriptPromise ||
    (window.grecaptcha ?
      Promise.resolve() :
      new Promise((resolve, reject) => {
        const url =
            'https://www.google.com/recaptcha/api.js' +
            `?onload=${encodeURIComponent(CALLBACK_NAME)}` +
            (locale ? `&hl=${encodeURIComponent(locale)}` : '')

        window[CALLBACK_NAME] = resolve

        const script = document.createElement('script')
        script.type = 'text/javascript'
        script.src = url
        script.onerror = err => {
          reject(
              new URIError(`The script ${err.target.src} is not accessible.`)
            )
        }

        document.head.appendChild(script)
      }))

  return loadScriptPromise
}
```

I like using the ternery operator here, so we should do that regardless. I'm not so sure about the `||` operator. If we ditch the `||` operator, then we're back to using an early return.

Reply:
Cool! With ternary it looks more clean.